modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/apps-search-results/ddm-template.ftl

```diff
@@ -57,62 +73,81 @@
 					<#assign
 						portalURL = portalUtil.getLayoutURL(themeDisplay)
 						productId = entry.getClassPK() + 1
-						product = restClient.get("/headless-commerce-admin-catalog/v1.0/products/" + productId + "?nestedFields=productSpecifications,attachments" )
-						productAttachments = product.attachments![]
-						productDescription = stringUtil.shorten(htmlUtil.stripHtml(product.description.en_US!""), 150, "..." )
+						product = restClient.get("/headless-commerce-delivery-catalog/v1.0/channels/"+ channelId +"/products/"+ productId +"?accountId=-1&nestedFields=productSpecifications,categories")
 						productSpecifications = product.productSpecifications![]
-						productURL=portalURL?replace("home", "p" ) + "/" + product.urls.en_US />
+						productURL=portalURL?replace("home", "p" ) + "/" + product.urls.en_US
+					/>
+
+					<#if product.name?has_content>
+						<#assign productName = product.name />
+					<#else>
+						<#assign productName = "" />
+					</#if>
+
+					<#if product.description?has_content>
+						<#assign productDescription = stringUtil.shorten(htmlUtil.stripHtml(product.description!""), 150, "...") />
+					<#else>
+						<#assign productDescription = "" />
+					</#if>
+
+					<#if product.urlImage?has_content>
+						<#assign
+							productThumbnail = product.urlImage
+							productImageParts = productThumbnail?split("/o/")
+						/>
+					<#else>
+						<#assign productThumbnail = "" />
+					</#if>
 
 					<a class="app-search-results-card bg-white border-radius-medium d-flex flex-column mb-0 p-3 text-dark text-decoration-none" href=${productURL}>
 						<div class="align-items-center card-image-title-container d-flex pb-3">
 							<div class="image-container rounded">
-								<#if productAttachments?has_content>
-									<#list productAttachments as attachmentFields>
-										<#list attachmentFields.customFields as field>
-											<#if (field.name=="App Icon" ) && (stringUtil.equals(field.customValue.data[0]?lower_case, "yes"))>
-												<#assign srcName = "/o/" + attachmentFields.src?keep_after("/o/") />
-
-												<img
-													alt=${product.name.en_US}
-													class="h-100 mw-100"
-													src="${srcName}" />
-											</#if>
-										</#list>
-									</#list>
-								</#if>
+								<img
+									alt=${productName}
+									class="app-search-image"
+									src="/o/${productImageParts[1]}"
+								/>
 							</div>
 
 							<div class="pl-2">
 								<div class="font-weight-semi-bold h2 mt-1">
-									${product.name.en_US}
+									${productName}
 								</div>
 								<#if productSpecifications?has_content>
-									<#assign productPriceModels = productSpecifications?filter(item -> item.specificationKey == "developer-name") />
+									<#assign productPriceModelsDeveloperName = productSpecifications?filter(item -> item.specificationKey == "developer-name") />
```

Comment:
This name is very strange for me. Could you change it to something like productDeveloperName?
```suggestion
									<#assign productDeveloperName = productSpecifications?filter(item -> item.specificationKey == "developer-name") />
```